archive/100-theorems-list/16_abel_ruffini.lean

```diff
@@ -0,0 +1,269 @@
+/-
+Copyright (c) 2021 Thomas Browning. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Thomas Browning
+-/
+import field_theory.abel_ruffini
+import analysis.calculus.local_extr
+/-!
+Construction of an algebraic number that is not solvable by radicals.
+
+The main ingredients are:
+ * `solvable_by_rad.is_solvable'` in `field_theory/abel_ruffini` :
+  an irreducible polynomial with an `is_solvable_by_rad` root has solvable Galois group
+ * `gal_action_hom_bijective_of_prime_degree'` in `field_theory/polynomial_galois_group` :
+  an irreducible polynomial of prime degree with 1-3 non-real roots has full Galois group
+ * `equiv.perm.not_solvable` in `group_theory/solvable` : the symmetric group is not solvable
+
+Then all that remains is the construction of a specific polynomial satisfying the conditions of
+  `gal_action_hom_bijective_of_prime_degree'`, which is done in this file.
+
+-/
+
+open polynomial polynomial.gal
+
+local attribute [instance] splits_ℚ_ℂ
+
```

Comment:
I would be inclined to explicitly define the family of polynomials you are working with:
```
def \Phi (R : Type*) [comm_semiring R] (a b : ℕ) : (X ^ 5 - C ↑a * X + C ↑b : polynomial R)
```
and then a lemma
```
@[simp] lemma map_Phi {R S : Type*} [comm_semiring R] [comm_semiring S] (f : R ->+* S) (a b : ℕ) :
  (\Phi R a b).map f = \Phi S a b :=
sorry
```
My impression from the proofs below, is that it would then be useful to have simp lemmas about `coeff 0`, `coeff 5` etc?